pkgs/development/compilers/opensycl/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, clang15Stdenv, fetchFromGitHub, llvmPackages_15, llvmPackages_rocm, rocm-device-libs, python3, rocm-runtime, cmake, boost, libxml2, libffi, makeWrapper, hip, rocmSupport ? false }:
+let
+  # depending on whether rocm support is enabled or not, we choose a different mkDerivation, postFixup routine and add an extra buildInput
+  mkDerivation = if rocmSupport then llvmPackages_rocm.rocmClangStdenv.mkDerivation else clang15Stdenv.mkDerivation;
+
+  # one could add --hipsycl-targets='omp;hip' here, but I guess this is best left to the user
+  postFixup =
+    if rocmSupport then ''
+      wrapProgram $out/bin/syclcc-clang \
+        --prefix PATH : ${lib.makeBinPath [ python3 ]} \
+        --add-flags "-L${llvmPackages_15.openmp}/lib" \
+        --add-flags "-I${llvmPackages_15.openmp.dev}/include" \
+        --add-flags "--rocm-device-lib-path=${rocm-device-libs}/amdgcn/bitcode"
+    '' else ''
+      wrapProgram $out/bin/syclcc-clang \
+        --prefix PATH : ${lib.makeBinPath [ python3 ]} \
+        --add-flags "-L${llvmPackages_15.openmp}/lib" \
+        --add-flags "-I${llvmPackages_15.openmp.dev}/include"
+    '';
+  extraBuildInput = if rocmSupport then [ rocm-runtime ] else [ llvmPackages_15.libclang.dev ];
+in
+mkDerivation {
+  name = "OpenSYCL";
+  version = "0.9.4";
+  src = fetchFromGitHub {
+    owner = "OpenSYCL";
+    repo = "OpenSYCL";
+    rev = "v0.9.4";
+    sha256 = "sha256-5YkuUOAnvoAD5xDKxKMPq0B7+1pb6hVisPAhs0Za1ls=";
+  };
+  nativeBuildInputs = [
+    cmake
+    boost
+    llvmPackages_15.openmp
+    llvmPackages_15.llvm
+  ] ++ lib.optionals rocmSupport [ hip ];
+  buildInputs = [
+    libxml2
+    libffi
+    makeWrapper
+  ] ++ extraBuildInput;
+  # opensycl makes use of clangs internal headers. It's cmake does not successfully discover them automatically on nixos, so we supply the path manually
+  cmakeFlags = [
+    "-DCLANG_INCLUDE_PATH=${llvmPackages_15.libclang.dev}/include"
+  ];
+  inherit postFixup;
+
+  meta = {
+    homepage = "https://github.com/OpenSYCL/OpenSYCL";
+    description = "Multi-backend implementation of SYCL for CPUs and GPUs";
+    maintainers = [ ];
```

Comment:
maintainers must be set

Reply:
Done, although I cannot help with anything but the AMD and CPU targets. Better than nothing though.